pkg/lib/cockpit-components-dialog.jsx

```diff
@@ -244,13 +248,13 @@ var DialogFooter = React.createClass({
  *  - footer (react element, top element should be of class modal-footer)
  *  - id optional, id that is assigned to the top level dialog node, but not the backdrop
  */
-var Dialog = React.createClass({
+var Dialog = createReactClass({
     propTypes: {
-        title: React.PropTypes.string.isRequired,
-        no_backdrop: React.PropTypes.bool,
-        body: React.PropTypes.element.isRequired,
-        footer: React.PropTypes.element.isRequired,
-        id: React.PropTypes.string,
+        title: PropTypes.string, // required, but show_modal_dialog provides it after initial rendering; TODO: fix
```

Comment:
I would prefer to fix the modal dialog in a follow-up.
Maybe by using patternfly-react's component for it but various Cockpit's use cases needs to be considered.